src/inputs/radio/radio.jsx

```diff
@@ -54,19 +69,222 @@ const defaultProps = {
     multi: false,
     name: null,
     onChange: null,
+    onKeyDown: null,
     pill: false,
     style: null,
     tabIndex: -1,
-    value: null,
+    value: '',
 };
 
-const isCheckedSingle = (c, i, isChecked) => (
-    c.id ? c.id === isChecked : i === isChecked
-);
+const isCheckedSingle = (id, index, isChecked) => {
+    return id === isChecked || index === isChecked;
+};
+
+const isCheckedMulti = (id, index, isChecked) => {
+    const newId = id || index;
+
+    return includes(isChecked, newId);
+};
+
+const styles = (theme) => {
+    const size = 22;
+    const sizeDot = 14;
 
-const isCheckedMulti = (c, i, isChecked) => {
-    const id = c.id ? c.id : i;
-    return includes(isChecked, id);
+    const inputStyle = {
+        display: 'none',
+        '&:checked +': {
+            '& $label::after': {
+                opacity: 1,
+            },
+        },
+    };
+
+    return {
+        isAlignedRight: {},
+        isDisabled: {},
+        isFluid: {},
+        isPill: {},
+        input: {
+            ...inputStyle,
+        },
+        label: {},
+        labelNotClickable: {},
+        root: {
+            display: 'inline-block',
+            marginRight: theme.spacing(2),
+            minHeight: size,
+            outline: 'none',
+            position: 'relative',
+            textAlign: 'left',
+            '&:focus': {
+                boxShadow: `0 0 0 1px ${theme.palette.active.primary}`,
+            },
+            '& .label': {
+                color: theme.palette.text.primary,
+                cursor: 'pointer',
+                display: 'block',
+                fontSize: 14,
+                position: 'relative',
+                '&::before, &::after': {
+                    content: '""',
```

Comment:
Should this just be an empty string (`''`)?  Or is this correct as-is?  Just checking 'cause it looks a little suspicious.  But if it's correct, all good.

Reply:
Correct as is and I agree it is ugly.

Reply:
Copy that.  C'est la vie.  Resolving.